lib/AuthTokenVerifier.ts

```diff
@@ -1,104 +1,133 @@
+import { JWT } from "@ew-did-registry/jwt";
+import { Keys } from "@ew-did-registry/keys";
+import {
+  IAuthentication,
+  IDIDDocument,
+  IPublicKey,
+} from "@ew-did-registry/did-resolver-interface";
+import { utils } from "ethers";
+import base64url from "base64url";
 
-import { JWT } from '@ew-did-registry/jwt';
-import { Keys } from '@ew-did-registry/keys';
-import { IAuthentication, IDIDDocument, IPublicKey } from "@ew-did-registry/did-resolver-interface";
+const { arrayify, recoverAddress, keccak256, hashMessage } = utils;
 
 export class AuthTokenVerifier {
+  private _jwt = new JWT(new Keys());
+  constructor(private readonly _didDocument: IDIDDocument) {}
 
-    private readonly privateKey: string
-    private readonly didDocument: IDIDDocument
-
-    constructor(
-        private readonly _privateKey: string,
-        private readonly _didDocument: IDIDDocument
-    ) {
-        this.didDocument = _didDocument,
-        this.privateKey = _privateKey
-    }
-
-    /**
+  /**
    * @description checks a token was signed by the issuer DID or a valid authentication delegate of the issuer
    *
    * @param token
-   * @param issuerDID
    *
-   * @returns {string} issuer DID or null
+   * @returns {string} Authenticated identity DID
    */
-    public async verify(token: string, issuerDID: string): Promise<string | null> {
-        if (await this.isAuthorized(token))
-            return issuerDID
-        return null
+  public async verify(token: string): Promise<string | null> {
+    if ((await this.isIdentity(token)) || (await this.isDelegate(token))) {
+      return this._didDocument.id;
     }
+    return null;
+  }
+  
+  /**
+   * @description Authenticates by token signed with Ethereum signature
+   * Jwt can't be verified against etherum address  
```

Comment:
@JGiter What does this line mean? Like the whole JWT can't be verified? Because we are using the signature from the JWT to recover the address.

Reply:
It authenticates whole token, but this token must be signed with ethereum signature, because verification of standard jwt requires public key